packages/sqlite_async/lib/src/native/database/connection_pool.dart

```diff
@@ -232,6 +239,66 @@ class SqliteConnectionPool with SqliteQueries implements SqliteConnection {
       await connection.refreshSchema();
     }
   }
+
+  Future<T> withAllConnections<T>(
+      Future<T> Function(
+              SqliteWriteContext writer, List<SqliteReadContext> readers)
+          block) async {
+    try {
+      _runningWithAllConnections = true;
+
+      final blockCompleter = Completer<T>();
+      final (write, reads) = await _lockAllConns<T>(blockCompleter);
```

Comment:
Can we move this into the `try` block to ensure this throwing wouldn't leave `_runningWithAllConnections` set?